report_qweb_field_options/models/ir_qweb.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2024-2025 Quartile (https://www.quartile.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class IrQweb(models.AbstractModel):
+    _inherit = "ir.qweb"
+
+    @api.model
+    def _get_field(
+        self, record, field_name, expression, tagName, field_options, values
+    ):
+        if values.get("report_type") == "pdf":
+            options_recs = self.env["qweb.field.options"].search(
+                [("res_model_name", "=", record._name), ("field_name", "=", field_name)]
```

Comment:
Please consider the company as well. Use `record.company_id` or `self.env.company`. Filter records where the company matches or is false.